lib/phoenix_html/link.ex

```diff
@@ -133,21 +133,44 @@ defmodule Phoenix.HTML.Link do
 
   All other options are forwarded to the underlying button input.
   """
-  def button(text, opts) do
-    {to, opts} = Keyword.pop(opts, :to)
-    {method, opts} = Keyword.pop(opts, :method, :post)
 
-    {form, opts} = form_options(opts, method, "button")
+  def button(opts) do
+    {to, form, opts} = extract_button_options(opts)
 
-    unless to do
-      raise ArgumentError, "option :to is required in button/2"
+    {do_option, opts} = pop_required_option(opts, :do, "option :do is required in button/1")
+
+    form_tag(to, form) do
+      Phoenix.HTML.Form.submit(opts, [do: {:safe, do_option}])
```

Comment:
This should not be needed. The EEx engine will automatically wrap it in `{:safe, ...}`. The only case it won't be wrapped is if you generating the buttons manually (say in Elixir) but then you must explicitly tag it as safe anyway (like in your tests).